File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3021_alice-and-bob-playing-flower-game/solutions/java/solution.java
class Solution {
    public long flowerGame(int n, int m) {
        long count = 0;
        for (int x = 1; x <= n; x++) {
            for (int y = 1; y <= m; y++) {
                if ((x + y) % 2 == 1) {
                    count++;
                }
            }
        }
        return count;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3021_alice-and-bob-playing-flower-game/solutions/cpp/solution.cpp
#include <algorithm>

class Solution {
public:
    long long flowerGame(int n, int m) {
        long long count = 0;
        for (int x = 1; x <= n; ++x) {
            for (int y = 1; y <= m; ++y) {
                if ((x + y) % 2 == 0 && (x - 1 <= y || y - 1 <= x)) {
                    count++;
                }
            }
        }
        return count;
    }
};
